source/config/__init__.py

```diff
@@ -1243,8 +1244,20 @@ def __setitem__(self, key, val):
 		except KeyError:
 			pass
 		else:
-			if val == curVal:
-				# The value isn't different, so there's nothing to do.
+			if (
+				(
+					# FeatureFlag overrides __eq___
+					not isinstance(val, FeatureFlag)
+					and val == curVal
+				)
+				or (
+					# FeatureFlag overrides __eq___
+					isinstance(val, FeatureFlag)
+					and val.value == curVal.value
+				)
+			):
+				# If the value is unchanged, do not update
+				# or mark the profile as dirty

```

Comment:
I don't really like this being coupled to the type implementation (FeatureFlag).
If I understand correctly, the "final" thing that makes two leaf nodes equal or not is their string representation when being written to the config file. Can this be used instead:

```python
if str(val) == str(curVal):

Reply:
Unfortunately `val` and `curVal` can be any type, including an `AggregatedSection`

```
			curVal = self.__getitem__(key, checkValidity=False)
```

Reply:
I'll have to check if it is a leaf before comparison